sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/serializer/GeneratedProtobufSerializer.scala

```diff
@@ -14,60 +14,64 @@
  * limitations under the License.
  */
 
-package kalix.serializer
+package kalix.springsdk.impl.serializer
 
 import scala.jdk.CollectionConverters.CollectionHasAsScala
 import scala.jdk.CollectionConverters.MapHasAsJava
 
 import com.fasterxml.jackson.dataformat.protobuf.ProtobufMapper
 import com.google.protobuf.ByteString
 import com.google.protobuf.Descriptors.FileDescriptor
+import com.google.protobuf.UnsafeByteOperations
 import com.google.protobuf.{ Any => JavaPbAny }
+import kalix.javasdk.impl.ByteStringEncoding
+import kalix.serializer.Serializer
+import kalix.springsdk.impl.serializer.GeneratedProtobufSerializer.KalixGeneratedProto
 
-object JacksonProtobufSerializer {
+object GeneratedProtobufSerializer {
 
-  final val KalixJacksonProto = "jackson.proto.kalix"
+  final val KalixGeneratedProto = "proto.kalix.io/"
```

Comment:
I'm not sure about `proto.kalix.io` naming. The goal is to say, that's proto defined by Kalix. Didn't want to pick `generated.proto.kalix.io`. Too long.

Reply:
@johanandren Here's the problem. Let's say I have the following the following path - note, this is intentionally contrived to be overly complex, and I don't ever expect that a user will have all these requirements compressed into the one single method, but I do expect that each individual requirement that this path has will be exhibited quite commonly, hence everything that this path will require will need to be supported:

`/projects/foo/services/8e2a1b3f-6832-42e6-aec3-abe636a9829c/generation/5/entity/shoppingcart`

So, the above path has a template spec that looks like this:

`/projects/{project_id}/services/{service_id}/generation/{generation}/entity/{entity_name}`

Now, let's say, in this case, the entity represents a generation of a service, so, the entity id must contain both the service id, and the generation. Meanwhile, while a service is part of a project, services can move between projects, so the project id is not part of the entity id. So, how do we instruct Kalix and Google HTTP transcoding to match the above path, extract the parameters out, correctly identify that the service id and generation are part of the entity key, all with a generic message? I don't think it's possible, what we need is a message that looks like this:

```protobuf
message MyRequest {
  string project_id = 1;
  string service_id = 2 [(kalix.field).entity_key = true];
  int32 generation = 3 [(kalix.field).entity_key = true];
  string entity_name = 4;
  string json_body = 5;
}

rpc MyMethod(MyRequest) returns (Any) {
  option (google.api.http) = {
    post: "/projects/{project_id}/services/{service_id}/generation/{generation}/entity/{entity_name}"
    body: "json_body"
  }
}
```

In order to support a generic request structure, we will need the following things:

1. Additional annotations added to the sdk protobufs to be able to specify how entity ids are extracted out of the path.
2. Explicit support for a generic wrapper type in the HTTP transcoding.

And while we could do that, we don't need to. We can generate message descriptors. It really isn't hard to do that, especially in the context of we're already generating a service descriptor and having to pass this configuration in some form on anyway, whether that comes out as additional options on a method descriptor that we've generated, or a generated message, it's not a real difference in complexity.

But, to add explicit support for all this in the proxy, that is additional complexity, especially when we introduce a second means for extracting entity ids out. It just seems like an unnecessary complexity to me when we already have a no additional complexity means to do it.

Reply:
As for working with it - the type is already dynamic in the proxy, adding support for a concrete type in the proxy actually increases complexity there since it means sometimes we are dealing with concrete types and sometimes dynamic, instead of always dynamic.

In the proxy, it's really no problem, working with dynamic protobuf types is not hard, I mean we do it with ease in the proxy. We will already have generated the descriptor to send the proxy, that same descriptor can be used to deserialise to DynamicMessage, which functions somewhat similar to a Map, and Maps are exactly what we'd be working with if we had a generic message, the primary difference is with a generic message, in our thing that extracts parameters to allow reflective invocation of the users methods, we'd have to have a reference to the name of the field to pull out of the map, whereas with a generated message, we'd have to have a reference to the id of the field. It's not harder. In both cases we'd have to deal with types and type conversions. It's not harder.